src/handle-data.js

```diff
@@ -31,10 +39,31 @@ export function getColumnInfo(qHyperCube, colIndex) {
       stylingInfo: info.qAttrExprInfo.map((expr) => expr.id),
       sortDirection: directionMap[info.qSortIndicator],
       dataColIdx: colIndex,
+      totalInfo,
     }
   );
 }
 
+/**
+ * Get the total head position of the table
+ *
+ * @param {Object} layout
+ * @returns the position as a string, it can be any of top, bottom or noTotals
+ */
+export function getTotalPosition(layout) {
+  let showTotal = true;
+  const hasOnlyMeasure = layout.qHyperCube.qDimensionInfo.length;
+  const hasGrandTotal = layout.qHyperCube.qGrandTotalRow.length;
+  const hasMeasure = layout.qHyperCube.qMeasureInfo.length;
+  const isTotalModeAuto = layout.totals?.show;
+
+  if (hasOnlyMeasure === 0 || !(hasMeasure > 0 && hasGrandTotal > 0)) showTotal = false;
+  if (!showTotal) return 'noTotals';
+  if (isTotalModeAuto || (!isTotalModeAuto && layout.totals.position === 'top')) return 'top';
+  if (!isTotalModeAuto && layout.totals.position === 'bottom') return 'bottom';
+  return 'noTotals';
```

Comment:
The names doesn' really make sense here. `has...` variables should be booleans by them selves. try this:

```suggestion
  const hasOnlyMeasure = layout.qHyperCube.qDimensionInfo.length === 0;
  const hasGrandTotal = layout.qHyperCube.qGrandTotalRow.length > 0;
  const hasMeasure = layout.qHyperCube.qMeasureInfo.length > 0;
  const isTotalModeAuto = layout.totals?.show;

  if (hasOnlyMeasure || !(hasMeasure && hasGrandTotal )) return 'noTotals';
  if (isTotalModeAuto || (!isTotalModeAuto && layout.totals.position === 'top')) return 'top';
  if (!isTotalModeAuto && layout.totals.position === 'bottom') return 'bottom';
  return 'noTotals';
```

Reply:
btw `if (hasOnlyMeasure || !(hasMeasure && hasGrandTotal )) return 'noTotals';` is the same as `if (hasOnlyMeasure || !hasMeasure || !hasGrandTotal) return 'noTotals';`